test/src/unit-conversions.cpp

```diff
@@ -734,235 +542,430 @@ TEST_CASE("value conversion")
+        }
+
+        SECTION("std::unordered_map<json::string_t, json>")
+        {
+            std::unordered_map<json::string_t, json> o = j;
+            CHECK(json(o) == j);
+        }
+
+        SECTION("std::unordered_multimap<json::string_t, json>")
+        {
+            std::unordered_multimap<json::string_t, json> o = j;
+            CHECK(json(o) == j);
+        }
+    }
+
+    SECTION("get an array (implicit)")
+    {
+        json::array_t a_reference{json(1),     json(1u),       json(2.2),
+                                  json(false), json("string"), json()};
+        json j(a_reference);
+
+        SECTION("json::array_t")
+        {
+            json::array_t a = j;
+            CHECK(json(a) == j);
+        }
+
+        SECTION("std::list<json>")
+        {
+            std::list<json> a = j;
+            CHECK(json(a) == j);
+        }
+
+        SECTION("std::forward_list<json>")
+        {
+            std::forward_list<json> a = j;
+            CHECK(json(a) == j);
+        }
+
+        SECTION("std::vector<json>")
+        {
+            std::vector<json> a = j;
+            CHECK(json(a) == j);
+        }
+
+        SECTION("std::deque<json>")
+        {
+            std::deque<json> a = j;
+            CHECK(json(a) == j);
+        }
+    }
+
+    SECTION("get a string (implicit)")
+    {
+        json::string_t s_reference{"Hello world"};
+        json j(s_reference);
+
+        SECTION("string_t")
+        {
+            json::string_t s = j;
+            CHECK(json(s) == j);
+        }
+
+#if defined(JSON_HAS_CPP_17)
+        SECTION("std::string_view")
+        {
+            std::string_view s = j.get<std::string_view>();
+            CHECK(json(s) == j);
+        }
+#endif
+
+        SECTION("std::string")
+        {
+            std::string s = j;
+            CHECK(json(s) == j);
+        }
+    }
+
+    SECTION("get a boolean (implicit)")
+    {
+        json::boolean_t b_reference{true};
+        json j(b_reference);
+
+        SECTION("boolean_t")
+        {
+            json::boolean_t b = j;
+            CHECK(json(b) == j);
+        }
+
+        SECTION("bool")
+        {
+            bool b = j;
+            CHECK(json(b) == j);
+        }
+    }
+
+#endif
+
+    SECTION("get a boolean (explicit)")
+    {
+        json::boolean_t b_reference{true};
+        json j(b_reference);
+
+        SECTION("boolean_t")
+        {
+            json::boolean_t b = j.get<json::boolean_t>();
+            CHECK(json(b) == j);
+        }
+
+        SECTION("uint8_t")
+        {
+            auto n = j.get<uint8_t>();
+            CHECK(n == 1);
+        }
+
+        SECTION("bool")
+        {
+            bool b = j.get<bool>();
+            CHECK(json(b) == j);
         }
 
         SECTION("exception in case of a non-number type")
         {
-            CHECK_THROWS_AS(json(json::value_t::null).get<json::number_integer_t>(),
+            CHECK_THROWS_AS(json(json::value_t::null).get<json::boolean_t>(),
```

Comment:
Why these changes to from `number_integer_t` to `boolean_t`?

Reply:
Oops, must be a leftover from last year